### PR TITLE
[11.0] FIX: fix leave number days when requested across years and public holidays are defined for each year

### PR DESCRIPTION
**Pull request by jarroyomorales**
Cherry-pick of https://github.com/OCA/hr/pull/671 into 11.0.

@guewen @alexey-pelykh @damdam-s

### CODE REVIEW

File: hr_contract_default_trial_length/static/description/index.html [hunk 1]
```diff
@@ -0,0 +1,453 @@
+<?xml version="1.0" encoding="utf-8" ?>
```
Comment (alexey-pelykh) [inline hr_contract_default_trial_length/static/description/index.html]: I don't think this file is needed?

**Review by alexey-pelykh — CHANGES_REQUESTED** [review]

**Comment by damdam-s** [comment]
@jarroyomorales seems there is a file that was not in #671 
Can you check please?

File: hr_contract_default_trial_length/static/description/index.html (hunk 1 above)
Comment (jarroyomorales) [reply hr_contract_default_trial_length/static/description/index.html]: Yes yes, already fixed it, did something wrong cherry-picking :sweat_smile:

**Comment by jarroyomorales** [comment]
> @jarroyomorales seems there is a file that was not in #671
> Can you check please?

Yes sorry, it should be fixed now!

File: hr_holidays_public/models/resource_calendar.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
+# Copyright 2017-2018 Tecnativa - Pedro M. Baeza
```
Comment (alexey-pelykh) [inline hr_holidays_public/models/resource_calendar.py]: Entirely new file? Very wrong cherry-pick?

**Review by alexey-pelykh — CHANGES_REQUESTED** [review]

**Review by alexey-pelykh — APPROVED** [review]

File: hr_holidays_public/models/resource_calendar.py (hunk 2 above)
Comment (jarroyomorales) [reply hr_holidays_public/models/resource_calendar.py]: Oops, the file name changed. Let's see now. It is my first time doing this.

**Comment by alexey-pelykh** [comment]
once CI is done, I'll merge

**Comment by alexey-pelykh** [comment]
@jarroyomorales it says that case is not covered by tests, please check

**Review by alexey-pelykh — CHANGES_REQUESTED** [review]
see test coverage

**Comment by jarroyomorales** [comment]
Yes, I'm working on it

**Comment by jarroyomorales** [comment]
I guess it works different in 11.0. I just ran the 2 new tests without applying any changes and they passed. So I think the bug doesn't apply to this version.

**Pull request closed without merging**